Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -1,13 +1,22 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     // TODO: make a struct and add validation and other needed methods (if needed)
     public class Credentials
     {
         public string Username;
         public string Password;
 
+        public string ToFile { get { return $"{Username},{Password}"; } }
+
         public Credentials(string username, string password)
         {
+            if (string.IsNullOrEmpty(username) || string.IsNullOrEmpty(password))
+            {
+                throw new ArgumentException();
```

Comment:
Please validate each separately and provide which argument was null. Use ``nameof`` for that purpose.
In general, when you throw an excpetion, you should give as much detail as possible to identify the cause.